libs/net/dal/Services/UserService.cs

```diff
@@ -123,6 +123,19 @@ public IPaged<User> Find(UserFilter filter)
             .FirstOrDefault(u => u.Id == id);
     }
 
+    /// <summary>
+    /// Find user by ID with minimal data (optimized for performance).
+    /// Only returns basic user information without related entities.
+    /// </summary>
+    /// <param name="id">User ID</param>
+    /// <returns>User with minimal data or null if not found</returns>
+    public User? FindByIdMinimal(int id)
```

Comment:
Nice to see we're thinking about the impact of each request and what is actually needed in each case.